### ISSUE TITLE
Valid schema returning `Invalid schema: Outermost schema can only be of type array, list, object or dict`

### ISSUE THREAD

**Issue by cfculhane** [bhch/react-json-form#91]
Thanks for the super useful library! 

Just having an issue today with a schema returning `Invalid schema: Outermost schema can only be of type array, list, object or dict`.  The example schema below parses fine on another playground, i.e. https://rjsf.netlify.app/

Example: 

```json
{
    "$ref": "#/$defs/OperationKwargFloatList",
    "$defs":
    {
        "OperationKwargFloatList":
        {
            "title": "OperationKwargFloatList",
            "type": "object",
            "properties":
            {
                "rules":
                {
                    "type": "array",
                    "items":
                    {
                        "$ref": "#/$defs/OperationKwargFloat"
                    },
                    "default":
                    []
                }
            },
            "required":
            []
        },
        "OperationKwargFloat":
        {
            "title": "OperationKwargFloat",
            "type": "object",
            "properties":
            {
                "type":
                {
                    "enum":
                    [
                        "OperationKwargFloat"
                    ]
                },
                "key":
                {
                    "type": "string",
                    "minLength": 1
                },
                "value":
                {
                    "type": "number"
                }
            },
            "required":
            [
                "type",
                "key",
                "value"
            ]
        }
    }
}
```

It seems like its to do with defrencing that top level `$ref`, because changing it to the below works fine:

```json
{
    "title": "OperationKwargFloatList",
    "type": "object",
    "properties":
    {
        "rules":
        {
            "type": "array",
            "items":
            {
                "$ref": "#/$defs/OperationKwargFloat"
            },
            "default":[]
        }
    },
    "required": [],
    "$defs":
    {
        "OperationKwargFloat":
        {
            "title": "OperationKwargFloat",
            "type": "object",
            "properties":
            {
                "type":
                {
                    "enum":
                    [
                        "OperationKwargFloat"
                    ]
                },
                "key":
                {
                    "type": "string",
                    "minLength": 1
                },
                "value":
                {
                    "type": "number"
                }
            },
            "required":
            [
                "type",
                "key",
                "value"
            ]
        }
    }
}
```

**Comment by bhch** [comment]
Thanks for reporting the issue. It will be fixed soon.

**Comment by cfculhane** [comment]
Great thank you for sorting that out so quickly!

**Comment by bhch** [comment]
Fix released in [react-json-form v2.13.0](https://github.com/bhch/react-json-form/releases/tag/v2.13.0) and [django-jsonform v2.21.0](https://github.com/bhch/django-jsonform/releases/tag/v2.21.0).

You can test it out in [the playground](https://bhch.github.io/react-json-form/playground/).

Thank you.
